hedera-node/src/main/java/com/hedera/services/state/expiry/classification/ClassificationWork.java

```diff
@@ -36,43 +36,71 @@
  */
 @Singleton
 public class ClassificationWork {
+    public static final List<MapAccessType> CLASSIFICATION_WORK = List.of(ACCOUNTS_GET);
+
     private final GlobalDynamicProperties dynamicProperties;
     private final EntityLookup lookup;
+    private final ExpiryThrottle expiryThrottle;
     private EntityNum lastClassifiedNum;
-    private MerkleAccount lastClassified = null;
-    private EntityNum payerForAutoRenewNum;
-    private MerkleAccount payerAccountForAutoRenew = null;
+    private MerkleAccount lastClassified;
+    private EntityNum payerNum;
+    private MerkleAccount payer;
 
     @Inject
     public ClassificationWork(
-            final GlobalDynamicProperties dynamicProperties, final EntityLookup lookup) {
+            final GlobalDynamicProperties dynamicProperties,
+            final EntityLookup lookup,
+            final ExpiryThrottle expiryThrottle) {
         this.dynamicProperties = dynamicProperties;
+        this.expiryThrottle = expiryThrottle;
         this.lookup = lookup;
     }
 
-    public ClassificationResult classify(final EntityNum candidateNum, final long now) {
+    public ClassificationResult classify(final EntityNum candidateNum, final Instant now) {
+        if (!expiryThrottle.allow(CLASSIFICATION_WORK, now)) {
+            return COME_BACK_LATER;
+        }
+
+        payer = null;
+        payerNum = null;
+        lastClassified = null;
         lastClassifiedNum = candidateNum;
-        if (!lookup.accountsContainsKey(lastClassifiedNum)) {
+
+        lastClassified = lookup.getImmutableAccount(lastClassifiedNum);
+        if (lastClassified == null) {
             return OTHER;
         } else {
-            lastClassified = lookup.getImmutableAccount(lastClassifiedNum);
+            final var longNow = now.getEpochSecond();
             final long expiry = lastClassified.getExpiry();
-            if (expiry > now) {
+            if (expiry > longNow) {
                 return OTHER;
             }
+
             final var isContract = lastClassified.isSmartContract();
-            if (lastClassified.getBalance() > 0) {
-                return isContract
-                        ? EXPIRED_CONTRACT_READY_TO_RENEW
-                        : EXPIRED_ACCOUNT_READY_TO_RENEW;
-            }
             if (lastClassified.isDeleted()) {
                 return isContract
                         ? DETACHED_CONTRACT_GRACE_PERIOD_OVER
                         : DETACHED_ACCOUNT_GRACE_PERIOD_OVER;
             }
+            if (isContract) {
+                if (!expiryThrottle.allow(CLASSIFICATION_WORK)) {
+                    return COME_BACK_LATER;
+                }
+                resolveClassifiedContractPayer();
```

Comment:
Wouldn't `autoRenew` account be the payer even if it is an account ?

Reply:
Right now, only contracts can have auto-renew accounts.